thyrel-api/thyrel-api/DataProvider/SessionDataProvider.cs

```diff
@@ -112,7 +119,10 @@ public async Task<Session> StartSession(int roomId)
 
             var playerDataProvider = new PlayerDataProvider(_holyDrawDbContext);
             var elementDataProvider = new ElementDataProvider(_holyDrawDbContext);
-            var addedSession = await Add(roomId, DateTime.Now.AddSeconds(duration), duration);
+
+            var playerCount = playerDataProvider.GetPlayersByRoom(roomId).Result.Count();
```

Comment:
ici il faut toujours utiliser await !!! et ici tu ne veux pas récup **tous** les players avec leur name etc puis ensuite les compter, tu as juste besoin de les compter.

Donc je te propose de directement allé taper sur le context ou de faire une méthode `GetPlayersCountByRoom`
```suggestion
            var playerCount = await _holyDrawDbContext.Player
                .Where(p => p.RoomId == roomId && p.IsConnected).CountAsync()
```

Reply:
okk 👍